chain/network/src/network_protocol/borsh.rs

```diff
@@ -130,7 +131,7 @@ pub(super) enum PeerMessage {
     Routed(Box<RoutedMessage>),
 
     /// Gracefully disconnect from other peer.
-    Disconnect,
+    Disconnect(ClosingReason),
```

Comment:
Don't touch borsh.rs, only add the new features to the protobuf format.
In particular, the change that you made here is backward incompatible.